### PR TITLE
Added implementations for get_agents_session, delete_agents_session and delete_agents

### PR DESCRIPTION
**Pull request by cheesecake100201**
Added implementations for not implemented methods for agents

### CODE REVIEW

File: llama_stack/apis/agents/agents.py [hunk 1]
```diff
@@ -443,11 +443,12 @@ async def get_agents_turn(
         self,
         agent_id: str,
         turn_id: str,
+        session_id: str
     ) -> Turn: ...
 
     @webmethod(route="/agents/step/get")
     async def get_agents_step(
-        self, agent_id: str, turn_id: str, step_id: str
+        self, agent_id: str, turn_id: str, step_id: str, session_id: str
```
Comment (raghotham) [inline llama_stack/apis/agents/agents.py:451]: Can you make sure we follow the same order of `agent_id: str, session_id: str, turn_id: str, step_id: str` in arguments? Also, are there corresponding changes needed in llama-stack-apps repo?

Comment (cheesecake100201) [reply llama_stack/apis/agents/agents.py]: Changed the order of parameters where ever possible and will check if the corresponding changes are required in llama stack apps repo

File: llama_stack/providers/impls/meta_reference/agents/agents.py [hunk 2]
```diff
@@ -138,24 +138,89 @@ async def _create_agent_turn_streaming(
...
+            turn = Turn(**turn)
+        except Exception as e:
+            raise ValueError(
+                f"Could not validate(?) Turns for {turn_id}"
+            ) from e
+        steps = turn.steps
+        for step in steps:
+            if step.step_id == step_id:
+                return AgentStepResponse(step=step)
+        raise ValueError("Provided step_id could not be found")
```
Comment (ashwinb) [inline llama_stack/providers/impls/meta_reference/agents/agents.py:177]: can you include the requested step id in this valueerror? 

```
ValueError(f"Provided {step_id=} could not be found")
```

File: llama_stack/providers/impls/meta_reference/agents/agents.py [hunk 3]
```diff
@@ -138,24 +138,89 @@ async def _create_agent_turn_streaming(
...
+        )
 
     async def delete_agents_session(self, agent_id: str, session_id: str) -> None:
-        raise NotImplementedError()
-
+        try:
+            await self.persistence_store.delete(f"session:{agent_id}:{session_id}")
+        except:
+            raise ValueError("Session Key Not Found")
     async def delete_agents(self, agent_id: str) -> None:
```
Comment (ashwinb) [inline llama_stack/providers/impls/meta_reference/agents/agents.py:222]: no newline here? did you run `pre-commit`?

**Review by ashwinb — CHANGES_REQUESTED** [review]
Thank you. I have some minor comments. It would be good to add a test or two? See the tests in `providers/tests/agents/test_agents.py` -- I would suggest making another file for just testing agent persistence (which is what this PR mostly targets.)

File: llama_stack/providers/impls/meta_reference/agents/agents.py (hunk 3 above)
Comment (cheesecake100201) [reply llama_stack/providers/impls/meta_reference/agents/agents.py:222]: No I did not. Will add a new line

File: llama_stack/providers/impls/meta_reference/agents/agents.py (hunk 2 above)
Comment (cheesecake100201) [reply llama_stack/providers/impls/meta_reference/agents/agents.py:177]: Done

**Comment by cheesecake100201** [comment]
> Thank you. I have some minor comments. It would be good to add a test or two? See the tests in `providers/tests/agents/test_agents.py` -- I would suggest making another file for just testing agent persistence (which is what this PR mostly targets.)

Will make another PR for that if thats okay with you.

File: llama_stack/providers/impls/meta_reference/agents/agents.py [hunk 4]
```diff
@@ -138,24 +138,89 @@ async def _create_agent_turn_streaming(
...
+            session_id=session_id,
+            turns=turns if turns else [],
+            started_at=session.started_at
+        )
 
     async def delete_agents_session(self, agent_id: str, session_id: str) -> None:
-        raise NotImplementedError()
+        try:
+            await self.persistence_store.delete(f"session:{agent_id}:{session_id}")
+        except:
```
Comment (ashwinb) [inline llama_stack/providers/impls/meta_reference/agents/agents.py:219]: you are catching all Exceptions here, that's not good. Can you just not catch and let the exception propagate upwards?

File: llama_stack/providers/impls/meta_reference/agents/agents.py [hunk 5]
```diff
@@ -138,24 +138,89 @@ async def _create_agent_turn_streaming(
...
-        raise NotImplementedError()
+    async def get_agents_turn(self, agent_id: str, session_id: str, turn_id: str) -> Turn:
+        turn = await self.persistence_store.get(f"session:{agent_id}:{session_id}:{turn_id}")
+        try:
+            turn = json.loads(turn)
+        except json.JSONDecodeError as e:
+            raise ValueError(
+                f"Could not JSON decode turn for {turn_id}"
+            ) from e
+        try:
```
Comment (ashwinb) [inline llama_stack/providers/impls/meta_reference/agents/agents.py:149]: i'd probably not add any try-catch here

File: llama_stack/providers/impls/meta_reference/agents/agents.py [hunk 6]
```diff
@@ -138,24 +138,89 @@ async def _create_agent_turn_streaming(
...
     ) -> AgentStepResponse:
-        raise NotImplementedError()
+        turn = await self.persistence_store.get(f"session:{agent_id}:{session_id}:{turn_id}")
+        try:
+            turn = json.loads(turn)
+        except json.JSONDecodeError as e:
+            raise ValueError(
+                f"Could not JSON decode turn for {turn_id}"
+            ) from e
+        try:
```
Comment (ashwinb) [inline llama_stack/providers/impls/meta_reference/agents/agents.py:167]: same thing here

File: llama_stack/providers/impls/meta_reference/agents/agents.py [hunk 7]
```diff
@@ -138,24 +138,89 @@ async def _create_agent_turn_streaming(
...
+        if turn_ids:
+            for turn_id in turn_ids:
+                turn = await self.persistence_store.get(f"session:{agent_id}:{session_id}:{turn_id}")
+                try:
+                    turn = json.loads(turn)
+                except json.JSONDecodeError as e:
+                    raise ValueError(
+                        f"Could not JSON decode session for {agent_id} and {session_id}"
+                    ) from e
+                try:
```
Comment (ashwinb) [inline llama_stack/providers/impls/meta_reference/agents/agents.py:202]: and here

**Review by ashwinb — APPROVED** [review]

**Comment by cheesecake100201** [comment]
Made formatting changes shown in pre-commit @ashwinb

**Comment by ashwinb** [comment]
Merged, thanks!

**Pull request merged**